tests/integration_tests/tests/agentless_tests/test_plugins.py

```diff
@@ -83,28 +93,86 @@ def test_delete_plugin_not_found(self):
             )
             self.assertEquals(404, e.status_code)
 
-    def _get_execution(self, workflow_id):
-        executions = self.client.executions.list(include_system_workflows=True)
-        ex_list = [e for e in executions if e.workflow_id == workflow_id]
-        self.assertEqual(len(ex_list), 1,
-                         msg='Expected to find 1 execution with workflow_id '
-                             '`{workflow_id}`, but found: '
-                             '{ex_list}'.format(
-                             workflow_id=workflow_id, ex_list=ex_list)
-                         )
-        return ex_list[0]
-
     def test_install_uninstall_workflows_execution(self):
         test_utils.upload_mock_plugin(TEST_PACKAGE_NAME, TEST_PACKAGE_VERSION)
 
-        ex = self._get_execution('install_plugin')
+        ex = self._get_latest_execution('install_plugin')
         self.wait_for_execution_to_end(ex)
 
         plugin = self.client.plugins.list()[0]
         self.client.plugins.delete(plugin.id)
 
-        ex = self._get_execution('uninstall_plugin')
+        ex = self._get_latest_execution('uninstall_plugin')
         self.wait_for_execution_to_end(ex)
 
         plugins = self.client.plugins.list()
         self.assertEqual(len(plugins), 0)
+
+
+class TestPluginsSystemState(AgentlessTestCase):
+    def test_installing_corrupted_plugin_doesnt_affect_system_integrity(self):
+        self._upload_plugin_and_assert_values(TEST_PACKAGE_NAME,
+                                              TEST_PACKAGE_VERSION,
+                                              plugins_count=0,
+                                              # Wagon has a bug that leaves 1
```

Comment:
I will remove the comments and put them in the commit msg, because it will be more clear there for future fixes

Reply:
I'd suggest putting them in a jira as well.
Either way, leaving them next to the code that would look wrong without them seems like a better idea to me than putting them in a commit message, meaning that someone looking at the code has to realise that the code is right, or go and do a git blame and hope there wasn't a refactor since a random comment about the code was left in a commit message.

Reply:
If the code had changed with those lines, so those the commit msg needs to add info about that.
And there is always git history :)